Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/ActorAsset.cpp

```diff
@@ -74,49 +73,30 @@ namespace AZ
                     const uint32_t originalVertex = sourceOriginalVertex[vertexIndex + vertexStart];
                     const uint32_t influenceCount = AZStd::GetMin<uint32_t>(
                         maxInfluencesPerVertex, static_cast<uint32_t>(sourceSkinningInfo->GetNumInfluences(originalVertex)));
-                    uint32_t influenceIndex = 0;
 
-                    AZStd::vector<uint32_t> localIndices;
-                    AZStd::vector<float> localWeights;
+                    uint32_t influenceIndex = 0;
                     for (; influenceIndex < influenceCount; ++influenceIndex)
                     {
                         EMotionFX::SkinInfluence* influence = sourceSkinningInfo->GetInfluence(originalVertex, influenceIndex);
-                        localIndices.push_back(static_cast<uint32_t>(influence->GetNodeNr()));
+                        if (influenceIndex % 2 == 0)
+                        {
+                            blendIndexBufferData.push_back(static_cast<uint32_t>(influence->GetNodeNr()) << 16);
+                        }
+                        else
+                        {
+                            blendIndexBufferData.back() |= static_cast<uint32_t>(influence->GetNodeNr());
+                        }
```

Comment:
Line 83 will truncate to 16 bits, but line 87 will not. If by some miracle someone cooks up an asset that overflows that 65535 boundary, it will overflow and corrupt the data in the other 16 bits. How about using `aznumeric_cast<uint16_t>` instead?
```suggestion
                        {
                            blendIndexBufferData.push_back(aznumeric_cast<uint16_t>(influence->GetNodeNr()) << 16);
                        }
                        else
                        {
                            blendIndexBufferData.back() |= aznumeric_cast<uint16_t>(influence->GetNodeNr());
                        }
```

But from the looks of this, the code here in ActorAsset.cpp is in full control over how the type for that `blendIndexBufferData` variable is constructed. Right now it is a vector of arrays of `uint32_t`s, and the low 16 and high 16 bits of those `uint32_t`s are filled individually. You could also make some struct of two `uint16_t`s instead, and then you wouldn't need to do the bit shifting manually to control which field is being set.